# Spatial Mapping of Single Cells via Correlation and Importance between Cells and Spots
The SM-CI method not only achieves precise spatial mapping of cell-to-spot but also provids deconvolution results for spatial spots by integrating ST data scRNA-seq data. This method adopts a dropout event handling strategy and establishes criteria for assessing the importance of cells and spots based on gene expression and spatial information. The overall workflow of SM-CI is showed in Fig. \ref{fig1}.
Firstly, a K-nearest neighbors imputation strategy is adopted to address dropout events in both ST and scRNA-seq data. Specifically, ST data is processed by leveraging gene expression patterns and spatial relationships between spots (see Algorithm 1), while scRNA-seq data is processed based on neighbor relationships within the same cell type (see Supplementary Algorithm S1). Secondly, We expand the dimensions of the spots in the ST data and randomly sample the scRNA-seq data to address the difference in cell counts between the two datasets. Furthermore, these two datasets are aligned by preserving common genes. Spatial position distance matrix, spot-spot Pearson correlation matrix, cell-cell Pearson correlation matrix, and cell-spot Pearson correlation matrix are constructed based on the aligned matrices. According to the proposed importance assessment criteria, each spot and cellâ€™s importance is calculated by using these matrices. Next, a cost matrix for a linear programming model is constructed based on spot and cell importance as well as the cell-spot correlation matrix. This enables the determination of one-to-one correspondence between cells and spots, thereby achieving the mapping of single cells to spatial spots. Finally, the spatial mapping accuracy is validated through various analyses, including reconstruction of spatial tissue, spatial distribution of cell counts, trajectory analysis, and spatial mapping of pseudotime values. 
